File: ios/Kurir/modules/auth/profile/ProfileModule.swift
//
//  ProfileModule.swift
//  Kurir
//
//  Created by Loren on 10/9/19.
//  Copyright © 2019 Crazybean Studio. All rights reserved.
//

import Foundation
import Mobilex

class ProfileModule: Module {
    override func inject() {
        // Wrapper
        factory(ProfileWrapper.self) { r in
            ProfileWrapper(repository: r.resolve(UsersRepository.self)!)
        }
        
        // Actor
        factory(ProfileActor.self) { (r, scene: ProfileScene) -> ProfileActor in
            ProfileActor(scene: scene, wrapper: r.resolve(ProfileWrapper.self)!)
        }
    }
}

File: android/src/main/java/au/com/crazybean/mobilex/kurir/modules/auth/profile/ProfileModule.kt
package au.com.crazybean.mobilex.kurir.modules.auth.profile

import au.com.crazybean.mobilex.kurir.extension.qualifier
import au.com.crazybean.mobilex.kurir.modules.base.ViewModel
import org.koin.androidx.viewmodel.dsl.viewModel
import org.koin.dsl.module


val profileModule = module {
    val qualifier = qualifier<ProfileWrapper>()

    // Wrapper via ViewModel
    viewModel(qualifier) {
        ViewModel(ProfileWrapper(get()))
    }

    // Actor
    factory { (scene: ProfileScene?) ->
        val viewModel = get<ViewModel<ProfileWrapper>>(qualifier)
        ProfileActor(scene, viewModel.wrapper)
    }
}